src/setuptools_scm/discover.py

```diff
@@ -2,11 +2,54 @@
 from .utils import trace, iter_entry_points
 
 
-def iter_matching_entrypoints(path, entrypoint):
-    trace("looking for ep", entrypoint, path)
-    for ep in iter_entry_points(entrypoint):
-        if os.path.exists(os.path.join(path, ep.name)):
-            if os.path.isabs(ep.name):
-                trace("ignoring bad ep", ep)
-            trace("found ep", ep)
-            yield ep
+def walk_potential_roots(root, search_parents=True):
+    """
+    Iterate though a path and each of its parents.
+    :param root: File path.
+    :param search_parents: If ``False`` the parents are not considered.
+    """
+
+    if not search_parents:
+        yield root
+        return
+
+    tail = root
+
+    while tail:
+        yield root
+        root, tail = os.path.split(root)
+
+
+def match_entrypoint(root, name):
+    """
+    Consider a ``root`` as entry-point.
+    :param root: File path.
+    :param name: Subdirectory name.
+    :return: ``True`` if a subdirectory ``name`` exits in ``root``.
+    """
+
+    if os.path.exists(os.path.join(root, name)):
+        if not os.path.isabs(name):
+            return True
+        trace("ignoring bad ep", name)
+
+    return False
+
+
+def iter_matching_entrypoints(root, entrypoint, config):
+    """
+    Consider different entry-points in ``root`` and optionally its parents.
```

Comment:
Also, since the last passing CI (it is a bit annoying that it does not run on every commit because I'm a first time contributor here) I cleaned up a bit here.

Reply:
@tdegeus i identified a bug in hg_git, i separated thins, release incoming